File: frontend/src/types/certificate.ts
/**
 * TypeScript types for course completion certificates.
 */
import type { LessonType } from "./section";

export interface CertificateResponse {
  id: string;
  user_id: string;
  course_id: string;
  certificate_uid: string;
  user_name: string;
  course_title: string;
  sections_count: number;
  lessons_count: number;
  completed_at: string;
}

/** Minimal lesson info returned in the detailed share view. */
export interface CertificateLessonBrief {
  title: string;
  lesson_type: LessonType;
}

/** Section with nested lessons for the detailed share view. */
export interface CertificateSectionBrief {
  title: string;
  lessons: CertificateLessonBrief[];
}

/** Extended certificate response with full course structure metadata. */
export interface CertificateDetailResponse extends CertificateResponse {
  course_description: string | null;
  sections: CertificateSectionBrief[];
}

File: backend/app/schemas/certificate.py
"""Pydantic schemas for course-completion certificates."""

import uuid
from datetime import datetime

from pydantic import BaseModel


class CertificateResponse(BaseModel):
    id: uuid.UUID
    user_id: uuid.UUID
    course_id: uuid.UUID
    certificate_uid: str
    user_name: str
    course_title: str
    sections_count: int
    lessons_count: int
    completed_at: datetime

    model_config = {"from_attributes": True}


# ── Detailed share view (public) ────────────────────


class LessonBrief(BaseModel):
    """Minimal lesson info for the public certificate detail page."""

    title: str
    lesson_type: str


class SectionBrief(BaseModel):
    """Minimal section info with its lessons for the public certificate detail page."""

    title: str
    lessons: list[LessonBrief]


class CertificateDetailResponse(CertificateResponse):
    """Extended certificate response that includes course structure metadata."""

    course_description: str | None = None
    sections: list[SectionBrief] = []
